nexus/tests/integration_tests/disks.rs

```diff
@@ -351,6 +352,115 @@ async fn test_disk_create_disk_that_already_exists_fails(
     disks_eq(&disks[0], &disk);
 }
 
+#[nexus_test]
+async fn test_disk_slot_assignment(cptestctx: &ControlPlaneTestContext) {
+    let client = &cptestctx.external_client;
+    DiskTest::new(&cptestctx).await;
+    create_org_and_project(client).await;
+    let nexus = &cptestctx.server.apictx().nexus;
+
+    let disk_names = ["a", "b", "c", "d"];
+    let mut disks = Vec::new();
+    for name in disk_names {
+        let disk = create_disk(&client, PROJECT_NAME, name).await;
+        disks.push(disk);
+    }
+
+    // Create an instance to which to attach the disks, then force it to stop
+    // to allow disks to be attached. There should be no disks attached
+    // initially.
+    let instance = create_instance(&client, PROJECT_NAME, INSTANCE_NAME).await;
+    let instance_id = &instance.identity.id;
+    let instance_next =
+        set_instance_state(&client, INSTANCE_NAME, "stop").await;
+    instance_simulate(nexus, &instance_next.identity.id).await;
+    let url_instance_disks =
+        get_instance_disks_url(instance.identity.name.as_str());
+    let listed_disks = disks_list(&client, &url_instance_disks).await;
+    assert_eq!(listed_disks.len(), 0);
+
+    let url_instance_attach_disk =
+        get_disk_attach_url(&instance.identity.id.into());
+
+    async fn get_disk_slot(
+        ctx: &ControlPlaneTestContext,
+        disk_id: Uuid,
+    ) -> i16 {
```

Comment:
So I think one consequence of the change I suggested earlier around the model type is that this would be a `u8`. I _think_ that's what we want, since we cast it to that internally in at least one place. It should be unsigned, for sure, and that seems an improvement in terms of the public API.